### PR TITLE
Setup DMS for elasticsearch

### PR DESCRIPTION
**Pull request by Renny4Real**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/PA-408

### *What is the problem we're trying to solve*

As part of implementing Elasticsearch for fuzzy address matching, we want to set up DMS to stream data into our Elasticsearch domain.

### *What changes have we introduced*

- Add terraform set up for DMS
- Add selection rules for DMS
- Add task settings for DMS

#### _Checklist_

- [ ] Added comments to the README or updated relevant documentation _(add a link to documentation)_, where necessary.
- [X] Code pipeline builds correctly

### *Follow up actions after merging PR*

Check DMS works in staging environment

### CODE REVIEW

File: terraform/staging/main.tf [hunk 1]
```diff
@@ -95,3 +99,39 @@ module "elasticsearch_db_staging" {
...
+  source_endpoint_identifier = "source-addresses-postgres"
+  source_db_engine_name      = "postgres"
+  source_db_port             = 5502
+  source_db_username         = data.aws_ssm_parameter.addresses_postgres_username.value
+  source_db_password         = data.aws_ssm_parameter.addresses_postgres_db_password.value
+  source_db_server           = data.aws_ssm_parameter.addresses_postgres_hostname.value
+  source_endpoint_ssl_mode   = "none"
+
+  //dms task set up
+  migration_type               = "full-load"
```
Comment (elcorbs) [inline terraform/staging/main.tf:132]: I think we will want `full-load-and-cdc` to continually capture changes

File: terraform/staging/selection_rules.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    },
+    {
+      "rule-type": "selection",
+      "rule-id": "3",
+      "rule-name": "3",
+      "object-locator": {
+        "schema-name": "%",
+        "table-name": "national_address"
+      },
+      "rule-action": "exclude"
```
Comment (elcorbs) [inline terraform/staging/selection_rules.json:32]: Do you need this if the whole schema hasn't been included?

File: terraform/staging/selection_rules.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      "rule-id": "4",
+      "rule-name": "4",
+      "object-locator": {
+        "schema-name": "%",
+        "table-name": "hackney_address"
+      },
+      "rule-action": "exclude"
+    },
+    {
+      "rule-type": "transformation",
```
Comment (elcorbs) [inline terraform/staging/selection_rules.json:45]: Is this the only column we don't query against?

**Review by elcorbs — APPROVED** [review]

File: terraform/staging/selection_rules.json (hunk 2 above)
Comment (Renny4Real) [reply terraform/staging/selection_rules.json:32]: Wasn't sure. 🤔 
Was following a few documentations for how to set this up but could be that these `exclude` rules aren't needed in this case.

File: terraform/staging/selection_rules.json (hunk 3 above)
Comment (Renny4Real) [reply terraform/staging/selection_rules.json:45]: Yeah. 
The other columns were either involved in querying or sorting the data. 
This was the only one I found that was not being used anywhere but I will check this again.

**Pull request merged**